src/main/java/cloud/orbit/actors/cluster/impl/RedisOrbitClient.java

```diff
@@ -68,40 +70,63 @@ public void run()
 
     public void subscribe(final String channelId, final MessageListener<Object> messageListener)
     {
+
         subscriptions.add(Pair.of(channelId, messageListener));
         if(isConnected)
         {
             redisClient.getTopic(channelId).addListener(messageListener);
         }
+
     }
 
     private void connectionTask()
     {
-        final boolean nowConnected = redisClient.getNodesGroup().pingAll();
-
-        if(!nowConnected)
+        try
         {
-            isConnected = false;
-        }
+            boolean nowConnected = false;
 
-        if(nowConnected && !isConnected)
-        {
-            boolean subscribedAll = true;
-            for(Pair<String, MessageListener<Object>> subscription : subscriptions)
+            try
             {
-                try
-                {
-                    redisClient.getTopic(subscription.getLeft()).removeAllListeners();
-                    redisClient.getTopic(subscription.getLeft()).addListener(subscription.getRight());
+                nowConnected = redisClient.getNodesGroup().pingAll();
+            }
+            catch (Throwable e)
+            {
+                logger.error("Unexpected error during redis ping", e);
+            }
+
+            if (!nowConnected)
+            {
+                isConnected = false;
+                logger.error("Redis is not connected: {}", redisClient.getNodesGroup().getNodes().stream()
+                        .map((x) -> x.getAddr().toString()).collect(Collectors.joining(",")));
+            }
 
-                } catch(Exception e)
+            if (nowConnected && !isConnected)
+            {
+                boolean subscribedAll = true;
+                for (Pair<String, MessageListener<Object>> subscription : subscriptions)
                 {
-                    subscribedAll = false;
-                    break;
+                    try
+                    {
+                        redisClient.getTopic(subscription.getLeft()).removeAllListeners();
+                        redisClient.getTopic(subscription.getLeft()).addListener(subscription.getRight());
+
+                    }
+                    catch (Exception e)
+                    {
+                        logger.warn("Could not resubscribe, this may be transient", e);
```

Comment:
Include context of what subscription it was trying to add a listener too?

Reply:
Context added.